Add Example 5 for updating existing package in autospec (#827)
clearlinux/clear-linux-documentation: source/guides/clear/autospec.rst

   if autospec is not able to automatically find the correct dependency on its
   own.

.. TODO: Document how to set up a license server for use with autospec

Example 5: Control file management
**********************************

TBD

Example 6: Update an existing package
*************************************

Generally, developers can expect a more streamlined approach to managing a large collection of packages with :command:`autospec`.

The |CL| team prefers to carry no patches and seeks to make the latest releases work. If we do need patches, we use :command:`autospec` to add, remove, or manage patches. The :command:`autospec` control files are integral to the patch management process.

Adding and submitting patches
=============================

* To add patches to |CL| upstream, follow our process
  in the |CL| `distribution repo contributing guide`_.

* To submit a patch to upstream, follow our process in

>   `contributing new packages`_.

Reviewer: For *submitting patches*, in addition to the above link, I think it will help to link to the section for contributing patches to existing packages as well: https://github.com/clearlinux/distribution/blob/master/contributing.md#contributing-to-an-existing-software-package

> If you maintain a downstream derivative of |CL| and you want to integrate the patch into your mix, follow the process in :ref:`mixer`.

Reviewer: ```suggestion
If you maintain a downstream derivative of |CL| and you want to integrate new or patched packages into your mix, follow the process in :ref:`mixer`.
```

Assuming you have followed the above process, :command:`autospec` has generated a new spec file. View the changes by following the next section.

Refresh a package and inspect
-----------------------------

#. To refresh the spec package and recreate RPM files, navigate to its

>    directory. In this case we use :command:`m4`.

Reviewer: ```suggestion
   directory. We will use the :command:`m4` package in this example.
```

#. Navigate to the top-level directory of the workspace

   .. code-block:: bash

      cd clearlinux

>    - where clearlinux is the top level of the tooling workspace

Reviewer: I think `clearlinux` should be emphasized in the markup on the above line, however you do that in RST :-)

#. Run the command:

     make clone_m4

#. Change directory to :file:`packages/m4`.

> #. This is last step of after making any modifications to control files or

Reviewer: ```suggestion
#. This is last step after making any modifications to control files or
```

   other files.

   .. code-block:: bash

      make autospec

> #. One example is to run :command:`git show m4.spec` to look at the diff to

Reviewer: ```suggestion
#. To view spec file changes, run :command:`git show m4.spec` to look at the diff to
```

   the spec file. The output shows:

    .. code-block:  console

       m4: Autospec creation for version 1.4.18

       diff --git a/m4.spec b/m4.spec
       index f76c78d..97b846a 100644
       --- a/m4.spec
       +++ b/m4.spec
       @@ -6,15 +6,14 @@
       #
       Name     : m4
       Version  : 1.4.18
       -Release  : 88
       +Release  : 89
       URL      : http://mirrors.kernel.org/gnu/m4/m4-1.4.18.tar.xz
       Source0  : http://mirrors.kernel.org/gnu/m4/m4-1.4.18.tar.xz
       -Source99 : http://mirrors.kernel.org/gnu/m4/m4-1.4.18.tar.xz.sig
       +Source1 : http://mirrors.kernel.org/gnu/m4/m4-1.4.18.tar.xz.sig
       Summary  : No detailed summary available

>        Group    : Development/Tools

Reviewer: In this code block, I suggest to indicate somehow that the output is abbreviated. Perhaps add ellipses on the final line?

> #. You can also run some of the following commands to view changes.

Reviewer: ```suggestion
#. The following commands provide a more complete view of the changes.
```
